File: determinant/ungolfed.py
def det(x):
    l = len(x)
    if l == 1:
        return x[0][0]
    return sum([(-1)**i*x[i][0]*det(minor(x,i+1,1)) for i in range(l)])

def minor(x,i,j):
    y = x[:]
    del(y[i-1])
    y=zip(*y)
    del(y[j-1])
    return zip(*y)

def main():
    x = [input()]
    for i in range(len(x[0])-1):
        x += [input()]
    print det(x)

if __name__ == '__main__':
    main()
    x = [[1,-4,9],[-6,7,3],[1,2,3]]

File: determinant/ungolfed.hs
import Data.List (transpose)

det :: Num a => [[a]] -> a
det [[a]]   = a
det m       = sum [(-1)^i*m!!i!!0*det(minor m (i+1) 1) | i <- [0..length m - 1]]

minor :: Num a => [[a]] -> Int -> Int -> [[a]]
minor m i j = transpose . remRow (j-1) . transpose $ remRow (i-1) m

remRow :: Num a => Int -> [[a]] -> [[a]]
remRow row m = ys ++ (tail zs)
    where (ys,zs) = splitAt row m

main = do
    putStrLn "aoeu"
